micrometer-observation/src/main/java/io/micrometer/observation/docs/ObservationDocumentation.java

```diff
@@ -159,13 +157,12 @@ default Observation observation(ObservationRegistry registry, @Nullable Observat
      * @param defaultConvention default convention that will be picked if there was
      * neither custom convention nor a pre-configured one via
      * {@link ObservationRegistry.ObservationConfig#observationConvention(GlobalObservationConvention)}
-     * @param context observation context
+     * @param contextSupplier observation context
      * @param registry observation registry
      * @return observation
      */
     default <T extends Observation.Context> Observation observation(@Nullable ObservationConvention<T> customConvention,
-            @NonNull ObservationConvention<T> defaultConvention, @NonNull T context,
-            @NonNull ObservationRegistry registry) {
```

Comment:
Unrelated to this change, but I removed the `@NonNull` annotations because they are redundant. We annotate the whole package with `@NonNullApi`.